mirrord/agent/src/steal/connection.rs

```diff
@@ -147,10 +150,86 @@ impl Client {
         }
 
         let framed = HTTP_FRAMED_VERSION.matches(&self.protocol_version);
+        let chunked = HTTP_CHUNKED_VERSION.matches(&self.protocol_version);
         let tx = self.tx.clone();
 
         tokio::spawn(async move {
-            if framed {
+            // Chunked data is preferred over framed data
+            if chunked {
+                // Send headers
+                tracing::error!("starting bg task");
+                let connection_id = request.connection_id;
+                let request_id = request.request_id;
+                let (
+                    Parts {
+                        method,
+                        uri,
+                        version,
+                        headers,
+                        ..
+                    },
+                    mut body,
+                ) = request.request.into_parts();
+                let _ = tx
+                    .send(DaemonTcp::HttpRequestChunked(ChunkedRequest::Start(
+                        HttpRequest {
+                            internal_request: InternalHttpRequest {
+                                method,
+                                uri,
+                                headers,
+                                version,
+                                body: (),
+                            },
+                            connection_id,
+                            request_id,
+                            port: request.port,
+                        },
+                    )))
+                    .await;
```

Comment:
```suggestion
                match body.next_frames(true).await {
                    Err(..) => return,
                    Ok(Frames { frames, is_last }) => {
                        let message = todo!(); // prepare first message
                        if tx.send(message).await.is_err() || is_last {
                            return;
                        }
                    }
                }
```
We'd like to send the ready frames withing the first mirrord-protocol message